homeassistant/components/roku/__init__.py

```diff
@@ -34,77 +26,60 @@
     extra=vol.ALLOW_EXTRA,
 )
 
-# Currently no attributes but it might change later
-ROKU_SCAN_SCHEMA = vol.Schema({})
+PLATFORMS = [MEDIA_PLAYER_DOMAIN, REMOTE_DOMAIN]
+SCAN_INTERVAL = timedelta(seconds=30)
 
 
-def setup(hass, config):
-    """Set up the Roku component."""
-    hass.data[DATA_ROKU] = {}
+async def async_setup(hass: HomeAssistant, config: Dict) -> bool:
+    """Set up the Roku integration."""
+    hass.data.setdefault(DOMAIN, {})
 
-    def service_handler(service):
-        """Handle service calls."""
-        if service.service == SERVICE_SCAN:
-            scan_for_rokus(hass)
-
-    def roku_discovered(service, info):
-        """Set up an Roku that was auto discovered."""
-        _setup_roku(hass, config, {CONF_HOST: info["host"]})
+    if DOMAIN in config:
+        for entry_config in config[DOMAIN]:
+            hass.async_create_task(
+                hass.config_entries.flow.async_init(
+                    DOMAIN, context={"source": SOURCE_IMPORT}, data=entry_config,
+                )
+            )
 
-    discovery.listen(hass, SERVICE_ROKU, roku_discovered)
+    return True
 
-    for conf in config.get(DOMAIN, []):
-        _setup_roku(hass, config, conf)
 
-    hass.services.register(
-        DOMAIN, SERVICE_SCAN, service_handler, schema=ROKU_SCAN_SCHEMA
-    )
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Roku from a config entry."""
+    try:
+        roku = Roku(entry.data[CONF_HOST])
+        roku_device_info = await hass.async_add_executor_job(roku.device_info)
+    except (RequestException, RokuException) as exception:
+        raise ConfigEntryNotReady from exception
+    except Exception as exception:  # pylint: disable=broad-except
+        raise ConfigEntryNotReady from exception
```

Comment:
Don't do this. If your code breaks for unknown reasons, let the integration break.